src/lib.rs

```diff
@@ -16,11 +25,77 @@
 use std::convert::From;
 use std::io::{Result, Write};
 use std::mem;
+use std::ops;
+use std::fmt;
```

Comment:
Yes, there would be a conflict actually: `io::Result` has already been imported. I bump into this issue with `fmt` quite often, and I usually make an exception for it, leaving it as a module.  I’d import `Index` from `ops`, merge `fmt` with `mem`, and sort.